### PR TITLE
Fix: Import Kit Issues - Warn users about missing PHP DOM extension [ED-12523]

### PR DESCRIPTION
**Pull request by Omerisra6**

### CODE REVIEW

File: app/modules/import-export/assets/js/hooks/use-kit.js [hunk 1]
```diff
@@ -54,7 +54,14 @@ export default function useKit() {
...
 			ajaxConfig.data.data = JSON.stringify( ajaxConfig.data.data );
 
-			return await runRequest( ajaxConfig );
+			return await runRequest( ajaxConfig ).catch( ( error ) => {
+				const response = 408 === error.status ? 'timeout' : error.responseJSON?.data;
+
+				setKitState( ( prevState ) => ( { ...prevState, ...{
+					status: KIT_STATUS_MAP.ERROR,
+					data: response || {},
+				} } ) );
```
Comment (StyleShit) [inline app/modules/import-export/assets/js/hooks/use-kit.js:63]:
```suggestion
				setKitState( ( prevState ) => ( { 
					...prevState,
					status: KIT_STATUS_MAP.ERROR,
					data: response || {},
				} ) );
```

File: app/modules/import-export/assets/js/shared/process-failed-dialog/process-failed-dialog.js [hunk 2]
```diff
@@ -60,6 +60,10 @@ const messagesContent = {
 		title: __( 'Couldn’t install the Kit', 'elementor' ),
 		text: __( 'The Kit includes plugins you don’t have permission to install. Contact your site admin to change your permissions.', 'elementor' ),
 	},
+	'no-domdocument': {
+		title: __( 'Couldn’t install the Kit', 'elementor' ),
```
Comment (StyleShit) [inline app/modules/import-export/assets/js/shared/process-failed-dialog/process-failed-dialog.js:64]:
```suggestion
		title: __( "Couldn't install the Kit", 'elementor' ),
```

File: app/modules/import-export/assets/js/hooks/use-kit.js (hunk 1 above)
Comment (Omerisra6) [reply app/modules/import-export/assets/js/hooks/use-kit.js:63]: Nice, Done.

File: app/modules/import-export/assets/js/shared/process-failed-dialog/process-failed-dialog.js (hunk 2 above)
Comment (Omerisra6) [reply app/modules/import-export/assets/js/shared/process-failed-dialog/process-failed-dialog.js:64]: Don`t you think it is better to keep the same way as the other errors?

File: app/modules/import-export/module.php [hunk 3]
```diff
@@ -43,6 +43,8 @@ class Module extends BaseModule {
 
 	const NO_WRITE_PERMISSIONS_KEY = 'no-write-permissions';
 
+	const NO_DOMDOCUMENT_KEY = 'no-domdocument';
```
Comment (StyleShit) [inline app/modules/import-export/module.php:46]: hmm, maybe?

```suggestion
	const NO_DOM_DOCUMENT_KEY = 'no-dom-document';
```

or if you wanna have the real reason

```suggestion
	const NO_LIBXML_EXT_KEY = 'no-libxml-ext';
```

Comment (Omerisra6) [reply app/modules/import-export/module.php:46]: I liked the second one, but we do require the DOMDocument extension.

File: app/modules/import-export/assets/js/shared/process-failed-dialog/process-failed-dialog.js [hunk 4]
```diff
@@ -60,6 +60,10 @@ const messagesContent = {
 		title: __( 'Couldn’t install the Kit', 'elementor' ),
 		text: __( 'The Kit includes plugins you don’t have permission to install. Contact your site admin to change your permissions.', 'elementor' ),
 	},
+	'no-domdocument': {
```
Comment (mark-elementor) [inline app/modules/import-export/assets/js/shared/process-failed-dialog/process-failed-dialog.js:63]:
```suggestion
	'domdocument-missing': {
```

File: app/modules/import-export/module.php (hunk 3 above)
Comment (mark-elementor) [inline app/modules/import-export/module.php:46]:
```suggestion
	const DOMDOCUMENT_MISSING = 'domdocument-missing';
```

File: app/modules/import-export/module.php [hunk 5]
```diff
@@ -489,6 +492,12 @@ private function ensure_writing_permissions() {
 		}
 	}
 
+	private function ensure_DOMDocument_exists() {
+		if ( ! class_exists( 'DOMDocument' ) ) {
+			throw new \Error( self::NO_DOMDOCUMENT_KEY );
```
Comment (mark-elementor) [inline app/modules/import-export/module.php:497]:
```suggestion
			throw new \Error( self::DOMDOCUMENT_MISSING );
```

Comment (Omerisra6) [reply app/modules/import-export/module.php:497]: Done.

File: app/modules/import-export/assets/js/shared/process-failed-dialog/process-failed-dialog.js (hunk 4 above)
Comment (Omerisra6) [reply app/modules/import-export/assets/js/shared/process-failed-dialog/process-failed-dialog.js]: Done.

**Review by mark-elementor — DISMISSED** [review]

File: app/modules/import-export/assets/js/shared/process-failed-dialog/process-failed-dialog.js [hunk 6]
```diff
@@ -60,6 +60,10 @@ const messagesContent = {
 		title: __( 'Couldn’t install the Kit', 'elementor' ),
 		text: __( 'The Kit includes plugins you don’t have permission to install. Contact your site admin to change your permissions.', 'elementor' ),
 	},
+	'domdocument-missing': {
+		title: __( 'Couldn’t install the Kit', 'elementor' ),
+		text: __( 'You must have DOMDocument installed', 'elementor' ),
```
Comment (mark-elementor) [inline app/modules/import-export/assets/js/shared/process-failed-dialog/process-failed-dialog.js:65]: Don't forget to update the text before merging :)

**Review by rotemee — DISMISSED** [review]

**Review by StyleShit — DISMISSED** [review]

**Review by StyleShit — APPROVED** [review]

**Pull request merged**